build.gradle.kts

```diff
@@ -141,21 +135,34 @@ sourceSets {
     }
 }
 
-kotlinDslPluginOptions {
-    experimentalWarning.set(false)
-}
-
 tasks {
-    withType<KotlinCompile>().configureEach {
-        kotlinOptions.jvmTarget = "1.8"
+    afterEvaluate {
+        // This needs to be in an afterEvaluate block,
+        // because otherwise KotlinDslCompilerPlugins would win, and override what we've set to Kotlin 1.8.
+        withType<KotlinCompile>().configureEach {
+            kotlinOptions.jvmTarget = JavaVersion.VERSION_1_8.toString()
+            // Supporting Gradle 5.0+ needs to use Kotlin 1.3.
+            // See https://docs.gradle.org/current/userguide/compatibility.html
+            kotlinOptions.apiVersion = "1.3"
+            // Theoretically we could use newer language version here,
+            // but sadly the @kotlin.Metadata created on the classes would be incompatible with Kotlin 1.3 consumers.
+            kotlinOptions.languageVersion = "1.3"
+            doFirst {
+                if (kotlinOptions.apiVersion == "1.3") {
+                    // Suppress "Language version 1.3 is deprecated and its support will be removed in a future version of Kotlin".
+                    kotlinOptions.freeCompilerArgs += "-Xsuppress-version-warnings"
+                } else {
+                    TODO("Remove -Xsuppress-version-warnings suppression, or change the condition to ${kotlinOptions.languageVersion}")
+                }
+            }
```

Comment:
Which part do you mean? 😅

 * I got `afterEvaluate { kotlinOptions }` simply by reading the code after being annoyed that without `afterEvaluate` it doesn't do anything.
 * The `doFirst` is something I started doing recently (a year or so), so that any hacks will auto-remove themselves when they're not necessary any more.
 * `apiVersion`/`languageVersion` is not a hack, the project has old compatible consumers, so need to back-compile.

Reply:
I recommend looking at individual commits, for example this built up in 3:
https://github.com/gradle-nexus/publish-plugin/pull/168/commits/c63aa3bb651ab438b2d7e9b08d70311d693336fc
https://github.com/gradle-nexus/publish-plugin/pull/168/commits/6f8bdfaa43b2a6808032509c8fcdb8cfd14c7fe2
https://github.com/gradle-nexus/publish-plugin/pull/168/commits/dd958d126e3e30ae0049a891f7a21aeb4915993f